pkg/operator/setup_manager.go

```diff
@@ -0,0 +1,94 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+
+	v1alpha1 "github.com/openshift/cert-manager-operator/api/operator/v1alpha1"
+	"github.com/openshift/cert-manager-operator/pkg/controller/istiocsr"
+	"github.com/openshift/cert-manager-operator/pkg/version"
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup-manager")
+)
+
+func init() {
+	ctrllog.SetLogger(klog.NewKlogr())
+
+	utilruntime.Must(clientscheme.AddToScheme(scheme))
+	utilruntime.Must(appsv1.AddToScheme(scheme))
+	utilruntime.Must(corev1.AddToScheme(scheme))
+	utilruntime.Must(rbacv1.AddToScheme(scheme))
+	utilruntime.Must(certmanagerv1.AddToScheme(scheme))
+	utilruntime.Must(v1alpha1.AddToScheme(scheme))
+	// +kubebuilder:scaffold:scheme
+}
+
+// Manager holds the manager resource for the istio-csr controller
+type Manager struct {
+	manager manager.Manager
+}
+
+// New creates a new manager.
+func New() (*Manager, error) {
+	setupLog.Info("setting up operator manager", "controller", istiocsr.ControllerName)
+	setupLog.Info("controller", "version", version.Get())
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Logger:                 ctrl.Log.WithName("operator manager"),
+		Scheme:                 scheme,
+		HealthProbeBindAddress: ":8081",
+		LeaderElection:         false,
+		LeaderElectionID:       "10c233b5.operator.openshift.io",
```

Comment:
```suggestion
		LeaderElectionID:       "10c233b5.certmanagers.operator.openshift.io",
```
to be consistent with the other API.


do you know what resources would this leader election end up using?
The other controller(s) we have issues a lease in the operator namespace.

Reply:
Here too lease is used, but in the istio namespace.

Reply:
`certmanagers` is the API name for for cert-manager operand, right, I will update it to below

```suggestion
		LeaderElectionID:       "10c233b5.istiocsrs.operator.openshift.io",
```

Reply:
doesn't seem fine to be using a `istiocsrs` suffix, as what I foresee is that in the future we would want to use controller-runtime for everything scoped under this operator (a very long endevaour rich in efforts which never made its way to reality, but hopefully soon after this..).

Reply:
More insights would be welcome, just to match you my thoughts:
The lease here is not controller specific, but specific to the controller-manager (one per operator?). Seemingly, leases are quite important to help avoid reconcile barriers, we don't want to break something later in the lifecycle. 

Say, hypothetically in between an upgrade:
- we've operator v1.18.0 with all stuff based on controller-runtime, using lease:`"10c233b5.certmanagers.operator.openshift.io"` and 
- old operator version v1.17.0 with lease ` "10c233b5.istiocsrs.operator.openshift.io"`
but we ought to run only one instance of this ever across upgrades and new versions whenever we default to controller-runtime too, this won't be able to break the tie atleast for the istio-csr controller's reconciles, for a breif period.

Reply:
suggestion incorporated.